### PR TITLE
refactor module.Close

### LINKED ISSUE #8: Refactor module.py, it is repetitive and brittle.
Module.Close is a big offender. (As this code is cleaned, it should be come easier to validate that all elements of the 'order' setting are actually existing sections. error.NoSuchSection should be created for this purpose.)

Module.Chunks will be able to be drastically simplified after Module.Close is cleaned up, I think.

### PR DESCRIPTION
**Pull request by martindemello**
Fixes #8 . The rest of the file looked fine, but if there are any other refactoring TODOs in there I can add them to this CL.

(I considered making each of the three sections marked off by `#-------------` a helper function but it didn't seem to add enough value)

### CODE REVIEW

File: vimdoc/module.py [hunk 1]
```diff
@@ -112,44 +112,47 @@ def Close(self):
...
-      self.Merge(commands)
-    if self.GetCollection(vimdoc.DICTIONARY) and 'dicts' not in self.sections:
-      dicts = Block(vimdoc.SECTION)
-      dicts.Local(id='dicts', name='Dictionaries')
-      self.Merge(dicts)
+    # ----------------------------------------------------------
+    # Add default sections.
+
+    # type : (id, name)
+    default_sections = {
```
Comment (dbarnett) [inline vimdoc/module.py]: I think you need a list of triples or an OrderedDict here to keep section ordering consistent.

File: vimdoc/module.py [hunk 2]
```diff
@@ -112,44 +112,47 @@ def Close(self):
...
+        vimdoc.FUNCTION: ('functions', 'Functions'),
+        vimdoc.EXCEPTION: ('exceptions', 'Exceptions'),
+        vimdoc.COMMAND: ('commands', 'Commands'),
+        vimdoc.DICTIONARY: ('dicts', 'Dictionaries'),
+        vimdoc.FLAG: ('config', 'Configuration'),
+        vimdoc.SETTING: ('config', 'Configuration'),
+      }
+
+    for k, (id, name) in default_sections.items():
+      if self.GetCollection(k) and (id not in self.sections):
```
Comment (dbarnett) [inline vimdoc/module.py]: Consider inverting this check to
```python
if not self.GetCollection(k) or id in self.sections:
  continue
```
for better legibility and less indent.

File: vimdoc/module.py [hunk 3]
```diff
@@ -112,44 +112,47 @@ def Close(self):
...
+        # Create the section if it does not exist.
+        block = Block(vimdoc.SECTION)
+        block.Local(id=id, name=name)
+        self.Merge(block)
+
+    # If any maktaba flags were documented, add a default configuration section
+    # to explain how to use them.
     if self.GetCollection(vimdoc.FLAG):
```
Comment (dbarnett) [inline vimdoc/module.py]: Comment to explain that the order is determined by the listing above (if that's actually true). If not, we'll want to change it to preserve the ordering that the preexisting version had.

File: vimdoc/module.py (hunk 2 above)
Comment (martindemello) [inline vimdoc/module.py]: it's just three lines, so i don't think the indentation matters much; mainly i find the forward condition easier to read than the inverse one.

File: vimdoc/module.py (hunk 3 above)
Comment (martindemello) [inline vimdoc/module.py]: done

**Pull request merged**